app/controllers/insured/group_selection_controller.rb

```diff
@@ -241,11 +257,18 @@ def term_or_cancel
     # @todo Refactor GroupSelectionController to implement the ideal solution. This is a temporary fix.
     # Redirects to root path unless RIDP is verified for the given market kind and family.
     (redirect_to(root_path) and return) unless ridp_verified?(hbx_enrollment.kind, hbx_enrollment.family)
-    @self_term_or_cancel_form = ::Insured::Forms::SelfTermOrCancelForm.for_post({enrollment_id: params.require(:hbx_enrollment_id), term_date: params[:term_date], term_or_cancel: params[:term_or_cancel]})
-
+    @self_term_or_cancel_form = ::Insured::Forms::SelfTermOrCancelForm.for_post(
+      enrollment_id: params.require(:hbx_enrollment_id),
+      term_date: params[:term_date],
+      term_or_cancel: params[:term_or_cancel],
+      cancellation_reason: params[:cancellation_reason]
+    )
     if @self_term_or_cancel_form.errors.present?
       flash[:error] = @self_term_or_cancel_form.errors.values.flatten.inject(""){|memo, error| "#{memo}<li>#{error}</li>"}
       redirect_to edit_plan_insured_group_selections_path(hbx_enrollment_id: params[:hbx_enrollment_id], family_id: params[:family_id])
+    elsif ENV['BS4_CONSUMER_FLOW_IS_ENABLED']
```

Comment:
used the ENV here to help keep DC as-is, open to thoughts surrounding it

Reply:
the general pattern we'll do for this kind of thing is pass in a `'bs4: "true"` param from the caller

Reply:
May need to review this whole commit other than just https://github.com/ideacrew/enroll/pull/4327/commits/adbbf95c4afa9407a11ed2e2d1a124d8ff98d22e#diff-9d1743baad3e5f99189e0df2ac697de176aa1765b4c998afc5429bc1c2570828R269-R289 but updated